File: Loenn/triggers/level_end_trigger.lua
local utils = require("utils")

return {
    name = "ScugHelper/LevelEndTrigger",
    depth = 100,
    placements = {
        {
            name = "trigger",
            data = {width = 32, height = 32, ShowCompleteScreen = true, ShowSpotlight = true, ScreenWipe = true},
        },
    },
    fillColor = {0, 0, 0, 0},
    outlineColor = {1, 1, 1}
}

File: Source/Entities/LevelEndTrigger.cs
using Microsoft.Xna.Framework;
using Monocle;
using Celeste.Mod.Entities;
using Celeste;
namespace Celeste.Mod.ScugHelper.Entities;

[CustomEntity("ScugHelper/LevelEndTrigger")]
public class LevelEndTrigger(EntityData data, Vector2 offset) : Trigger(data, offset) {
    public bool showCompleteScreen = data.Bool("ShowCompleteScreen", true);
    public bool spotlightWipe = data.Bool("ShowSpotlight", true);
    public bool screenWipe = data.Bool("ScreenWipe", true);

    private bool Lock = false;

    public override void OnEnter(Player player) {
        if (Lock) return;
        EndLevel();
        Lock = true;
    }

    private void EndLevel() {
        Level level = SceneAs<Level>();
        level.CompleteArea(spotlightWipe: spotlightWipe, skipScreenWipe: !screenWipe, skipCompleteScreen: !showCompleteScreen);
    }
}
